rackcity/views/it_model_views.py

```diff
@@ -79,38 +104,60 @@ def model_modify(request):
     Modify an existing model
     """
     data = JSONParser().parse(request)
-    failure_message = ""
     if 'id' not in data:
-        failure_message += "Must include id when modifying a model. "
-    else:
-        id = data['id']
-        try:
-            existing_model = ITModel.objects.get(id=id)
-        except ObjectDoesNotExist:
-            failure_message += "No existing model with id="+str(id)+". "
-        else:
-            try:
-                validate_model_height_change(data, existing_model)
-            except LocationException as error:
-                failure_message = "Height change of model causes conflicts. " + \
-                    str(error)
-                return JsonResponse(
-                    {"failure_message": failure_message},
-                    status=HTTPStatus.NOT_ACCEPTABLE
-                )
-            for field in data.keys():
-                setattr(existing_model, field, data[field])
-            try:
-                existing_model.save()
-                log_action(request.user, existing_model, Action.MODIFY)
-                return HttpResponse(status=HTTPStatus.OK)
-            except Exception as error:
-                failure_message = failure_message + str(error)
-    failure_message = "Request was invalid. " + failure_message
-    return JsonResponse({
-        "failure_message": failure_message
-    },
-        status=HTTPStatus.NOT_ACCEPTABLE
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value + GenericFailure.UNKNOWN.value,
+                "errors": "Must include 'id' when modifying a model"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    id = data['id']
+    try:
+        existing_model = ITModel.objects.get(id=id)
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "Model" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing model with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    try:
+        validate_model_height_change(data, existing_model)
+    except LocationException as error:
+        location_failure = "Height change of model causes conflicts. " + \
+            str(error)
```

Comment:
str(error) is ok for failure message here because they are nice looking errors that Ben wrote up in LocationException